drivers/network/tcpip/tcpip/fileobjs.c

```diff
@@ -581,27 +589,27 @@ NTSTATUS FileCloseAddress(
   PTDI_REQUEST Request)
 {
   PADDRESS_FILE AddrFile = Request->Handle.AddressHandle;
-  KIRQL OldIrql;
+  PCONNECTION_ENDPOINT Listener;
 
   if (!Request->Handle.AddressHandle) return STATUS_INVALID_PARAMETER;
 
-  LockObject(AddrFile, &OldIrql);
+  LockObject(AddrFile);
 
   if (InterlockedDecrement(&AddrFile->Sharers) != 0)
   {
       /* Still other guys have open handles to this, so keep it around */
-      UnlockObject(AddrFile, OldIrql);
+      UnlockObject(AddrFile);
       return STATUS_SUCCESS;
   }
 
   /* We have to close this listener because we started it */
-  if( AddrFile->Listener )
+  Listener = AddrFile->Listener;
```

Comment:
Sure

Reply:
And there you get this piece of code crashing: https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/lib/drivers/ip/transport/tcp/event.c;h=1d9918c821fe4bb6d0a829f0b246d6c540afabf2;hb=HEAD#l277

It can't be more broken than it currently is...